### DISCUSSION TITLE
Same name of a client is not allowed

### DISCUSSION THREAD

**Discussion by vatsalbhatia** [madrecha/jamku#830] [QA] [answered]
Client Name 1 = Vikram Singh
Client Name 2 = Vikram Singh

Jamku does not allow this to happen, to continue using this.... we need to enter a dummy character into it. this is not fair.

Two person of same name could be there.

Jamku could issue a warning before creating, at best.

Pl allow it.

**Comment by adarshmadrecha** [comment, marked as answer]
The primary key in Client is combination of following fields.
* First Name
* Last Name
* Business Name

If all the fields are exactly same, you will get error "Duplicate Error".

Reason behind this behaviour.
1. A simple (Not perfect but effective) duplicity check for Client Master creation
2. When user has to select a client in a selection field, he should be able to uniquely identify the client from these fields.

Practically this is not an hinderance because, we need a way to uniquely identify the client. Thus we would add some sort of uniqueness to the client name.
Examples to add uniqueness:
1. Location of the Client
2. Business name / Shop name of the Client
3. Reference of the client
4. Family Name of the Client
